client/jetpack-onboarding/main.jsx

```diff
@@ -31,16 +37,27 @@ class JetpackOnboardingMain extends React.PureComponent {
 	// TODO: Add lifecycle methods to redirect if no siteId
 
 	render() {
-		const { recordJpoEvent, siteId, siteSlug, stepName, steps } = this.props;
+		const {
+			isRequestingSettings,
+			recordJpoEvent,
+			settings,
+			siteId,
+			siteSlug,
+			stepName,
+			steps,
+		} = this.props;
 		return (
 			<Main className="jetpack-onboarding">
+				<QueryJetpackOnboardingSettings siteId={ siteId } />
```

Comment:
This will throw an error if we try accessing the onboarding flow with a slug that doesn't belong to a site we have credentials for. That won't be a problem if we land #21563; otherwise we might want to address it either in the query component or inline here.

Reply:
I was going to review #21536 first, but for the sake of reducing interdependence, it might make sense for this PR (or `QueryJetpackOnboardingSettings` in general) to be more resilient to that kind of situation...

Reply:
92faa16